### PR TITLE
Automatically format code while building the Project 

### PR DESCRIPTION
**Pull request by Liaojinghui**
This work is done by @Qiao-Jin, @superboyiii, and I.

### CODE REVIEW

File: .github/workflows/main.yml [hunk 1]
```diff
@@ -21,8 +21,8 @@ jobs:
         dotnet-version: ${{ env.DOTNET_VERSION }}
     - name: Check format
       run: |
-        dotnet tool install --version 5.0.142902 --tool-path ./ dotnet-format --add-source https://pkgs.dev.azure.com/dnceng/public/_packaging/dotnet-tools/nuget/v3/index.json
-        ./dotnet-format --check -v diagnostic
+        dotnet tool install -g dotnet-format --version 6.0.* --add-source https://pkgs.dev.azure.com/dnceng/public/_packaging/dotnet-tools/nuget/v3/index.json
```
Comment (shargon) [inline .github/workflows/main.yml:24]: I think that it's better to use a deterministic version for UT

**Review by shargon — APPROVED** [review]

**Review by erikzhang — APPROVED** [review]

**Comment by erikzhang** [comment]
Can it be done locally?

**Comment by Liaojinghui** [comment]
> Can it be done locally?

We can make the format command only works in debug mode, and set the ci in release mode.

**Comment by Liaojinghui** [comment]
Added an `env` variable to indicate the building environment instead.

**Comment by erikzhang** [comment]
@Liaojinghui I tested it. If this solution is adopted, the `dotnet-format` global tool must be installed in the development environment, otherwise the project cannot be built.

**Comment by erikzhang** [comment]
image

**Comment by Liaojinghui** [comment]
> image
> 
> 

Isn't that we all need to install 'dotnet format' to make sure our code is in proper style? Might try to add some checking logic.

**Comment by erikzhang** [comment]
I usually don't install `dotnet-format`, but use `Ctrl K + D` to format the code in Visual Studio.

**Comment by ZhangTao1596** [comment]
I use `auto format on save` in Visual Studio Code.

**Comment by Liaojinghui** [comment]
> I use `auto format on save` in Visual Studio Code.

Buddy, this can not cover all cases. I would say that most guys are using VS.

**Pull request closed without merging**